### PR TITLE
gromacs: update application description

### PR DESCRIPTION
**Pull request by al42and**
## Description of changes

Update the license and website for GROMACS:

- The website was updated ~1 year ago; can be verified by checking the URLs.
- The license was changed many years ago from GPL2 to LGPL 2.1; see the same "about" page or the [COPYING](https://gitlab.com/gromacs/gromacs/-/blob/main/COPYING?ref_type=heads) file.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
This is a bit complicated, because the licensing info is not retroactive.

Have you tried to update the package?

**Comment by al42and** [comment]
I'm not a NixOS user; I initially wanted to open an issue, but a PR seemed to be an easier way to describe the requested change :) I can close this and open an issue instead if that's preferred.

If I understand you correctly, the problem is that updating an already installed package won't change its license metadata. GROMACS switched to LGPL >10 years ago (https://github.com/gromacs/gromacs/commit/e6cd064af2032c3e2c9b05dc95dd4651700d5691#diff-400d38e5cfee181230373a8b02d38f50271a5bf62c62410382efddb8e8b19e22); presumably, few enough people are troubled by the wrong license data, so I see little reason to overcomplicate things.

File: pkgs/applications/science/molecular-dynamics/gromacs/default.nix [hunk 1]
```diff
@@ -72,7 +72,7 @@ in stdenv.mkDerivation rec {
 
   meta = with lib; {
     homepage = "https://www.gromacs.org";
-    license = licenses.gpl2;
+    license = licenses.lgpl21;
```
Comment (wegank) [inline pkgs/applications/science/molecular-dynamics/gromacs/default.nix:75]:
```suggestion
    license = licenses.lgpl21Plus;
```

**Review by wegank — APPROVED** [review]

**Comment by AndersonTorres** [comment]
> I can close this and open an issue instead if that's preferred.

No need to open an issue.
The purpose of an issue is distinct from a PR.

> 
> If I understand you correctly, the problem is that updating an already installed package won't change its license metadata.

The `meta` field must be accurate. 

If I package a software whose 1.0 is GPL but the 1.1 is BSD, it makes no sense to change the license of 1.0.

On the other hand, if you are merely fixing up a wrong metainfo, say the 1.1 is BSD but we marked GPL, then the change is most welcome.

> presumably, few enough people are troubled by the wrong license data

Well, nix provides access to each and every past commit of nixpkgs. This is useful for data hoarding and [software preservation](https://www.softwareheritage.org/2020/06/18/welcome-nixpkgs/)

> , so I see little reason to overcomplicate things.

Fair. I want only to know if the info was accurate.

**Review by AndersonTorres — APPROVED** [review]
Ping me 36h to merge

**Comment by al42and** [comment]
> If I package a software whose 1.0 is GPL but the 1.1 is BSD, it makes no sense to change the license of 1.0.
> 
> On the other hand, if you are merely fixing up a wrong metainfo, say the 1.1 is BSD but we marked GPL, then the change is most welcome.

I see, thanks for the explanation. I'm just fixing the metainfo here which has been incorrect since 24c79d66261d0d4a71feb14bc57a59e97f071ab8.

File: pkgs/applications/science/molecular-dynamics/gromacs/default.nix [hunk 2]
```diff
@@ -72,7 +72,7 @@ in stdenv.mkDerivation rec {
 
   meta = with lib; {
     homepage = "https://www.gromacs.org";
-    license = licenses.gpl2;
```
Comment (AndersonTorres) [inline pkgs/applications/science/molecular-dynamics/gromacs/default.nix:75]: 1. Squash the two commits into only one
2. Reword the commit:

```
gromacs: update meta

- license is now lgpl21Plus
- accurate longDescription
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
.

File: pkgs/applications/science/molecular-dynamics/gromacs/default.nix (hunk 2 above)
Comment (al42and) [reply pkgs/applications/science/molecular-dynamics/gromacs/default.nix:75]: Done

**Pull request merged**